src/futils.c

```diff
@@ -21,32 +21,41 @@ int git_futils_mkpath2file(const char *file_path, const mode_t mode)
 		GIT_MKDIR_PATH | GIT_MKDIR_SKIP_LAST | GIT_MKDIR_VERIFY_DIR);
 }
 
+static void git_buf_put_rand(git_buf *buf, size_t chars)
+{
+	size_t i;
+	const char* alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+
+	for (i = 0; i < chars; ++i) {
+		git_buf_putc(buf, alphabet[rand() % strlen(alphabet)]);
+	}
+}
+
+// Reimplementation of mkstemp, except with a umask-modified mode, instead of 0600.
 int git_futils_mktmp(git_buf *path_out, const char *filename, mode_t mode)
 {
+	// glibc tries 238328 times, OpenBSD tries INT_MAX times.
+	unsigned int tries = INT_MAX;
 	int fd;
-	mode_t mask;
-
-	p_umask(mask = p_umask(0));
+	const int open_flags = O_RDWR | O_CREAT | O_EXCL | O_BINARY | O_CLOEXEC;
 
-	git_buf_sets(path_out, filename);
-	git_buf_puts(path_out, "_git2_XXXXXX");
+	while (--tries > 0) {
+		git_buf_sets(path_out, filename);
+		git_buf_puts(path_out, "_git2_");
+		git_buf_put_rand(path_out, 8);
 
-	if (git_buf_oom(path_out))
-		return -1;
+		if (git_buf_oom(path_out))
+			return -1;
 
-	if ((fd = p_mkstemp(path_out->ptr)) < 0) {
-		git_error_set(GIT_ERROR_OS,
-			"failed to create temporary file '%s'", path_out->ptr);
-		return -1;
-	}
+		if ((fd = p_open(path_out->ptr, open_flags, mode)) == -1) {
```

Comment:
MInor formatting nit: we'd prefer no braces around a single line.

Reply:
Done.